rs/modelator/src/model/language/tla/json/parser.rs

```diff
@@ -1,65 +1,126 @@
 use nom::{
     branch::alt,
-    bytes::complete::{is_not, tag, take_while},
-    character::complete::{char, digit1, multispace0, satisfy},
-    combinator::{complete, cut, map, opt, recognize, value},
+    bytes::complete::{tag, take_while},
+    character::complete::{char, digit1, multispace0, multispace1, satisfy},
+    combinator::{complete, cut, eof, map, not, opt, peek, success, value},
     multi::{many1, separated_list0, separated_list1},
-    sequence::{delimited, pair, preceded, separated_pair},
+    sequence::{delimited, pair, preceded, separated_pair, terminated},
     IResult,
 };
 
-use serde_json::Value as JsonValue;
+use serde_json::{json, Value as JsonValue};
 
 pub(crate) fn parse_state(i: &str) -> IResult<&str, JsonValue> {
     map(
         preceded(
             opt(delimited(multispace0, complete(tag("/\\")), multispace0)),
             separated_list0(
                 delimited(multispace0, complete(tag("/\\")), multispace0),
-                parse_var,
+                parse_assignment,
             ),
         ),
-        |value| JsonValue::Object(value.into_iter().collect()),
+        |pairs| {
+            let j = JsonValue::Object(pairs.into_iter().collect());
+            println!("{}", serde_json::to_string_pretty(&j).unwrap());
+            j
+        },
     )(i)
 }
 
-fn parse_var(i: &str) -> IResult<&str, (String, JsonValue)> {
+fn parse_assignment(i: &str) -> IResult<&str, (String, JsonValue)> {
     delimited(
         multispace0,
         separated_pair(
-            parse_identifier,
+            parse_identifier_as_string,
             delimited(multispace0, char('='), multispace0),
-            parse_any_value,
+            parse_anything,
         ),
         multispace0,
     )(i)
 }
 
 // TODO: what else can TLA var idenfitiers have?
-fn parse_identifier(i: &str) -> IResult<&str, String> {
+fn parse_identifier_as_string(i: &str) -> IResult<&str, String> {
     map(
         many1(satisfy(|c| c.is_alphanumeric() || "_-".contains(c))),
         |value| value.into_iter().collect(),
     )(i)
 }
 
-fn parse_identifiers_as_values(i: &str) -> IResult<&str, JsonValue> {
-    map(parse_identifier, JsonValue::String)(i)
+fn parse_identifier(i: &str) -> IResult<&str, JsonValue> {
+    map(parse_identifier_as_string, |x| json!(x))(i)
+}
+
+fn parse_any_value_consuming(i: &str) -> IResult<&str, JsonValue> {
+    alt((
+        parse_bool,
+        parse_number,
+        parse_string,
+        parse_identifier,
+        parse_set,
+        parse_sequence,
+        parse_record,
+    ))(i)
+}
+
+// TODO: add more TLA+ operators? support prefix operators
+fn parse_operator(i: &str) -> IResult<&str, JsonValue> {
+    map(alt((tag("\\in"), tag("\\notin"))), |x| {
```

Comment:
Right now, I have only these two dummy operators.
Would it be good to add other operators?